dropshot/src/server.rs

```diff
@@ -290,6 +300,22 @@ async fn http_request_handle_wrap<C: ServerContext>(
         "uri" => format!("{}", request.uri()),
     ));
     trace!(request_log, "incoming request");
+    dropshot_request_start!(|| {
+        let uri = request.uri();
+        crate::RequestInfo {
+            id: request_id.parse().unwrap(),
```

Comment:
Maybe we should have `generate_request_id()` return a Uuid rather than a String so that we don't have to parse and unwrap here.  (It's a little scary to unwrap inside an IS_ENABLED block.)

Reply:
@davepacheco Coming back to this. There's a note in the docstring for `generate_request_id` discussing options for including other information in the ID, such as the service name. That'd only make sense if this is a String, not a Uuid. Is that still a goal? If not, I can update the function to return a Uuid and avoid any further parsing/unwrapping.

Reply:
A simpler option here is to change the `RequestInfo::id` field to also be a string. I'll go with that for now.

Reply:
Done in b52dc29